pkg/server/endpoints/registration/handler.go

```diff
@@ -285,6 +287,36 @@ func (h *Handler) ListBySpiffeID(ctx context.Context, request *registration.Spif
 	}, nil
 }
 
+//ListAllEntriesWithPages retrieves all registered entries with pagination.
+func (h *Handler) ListAllEntriesWithPages(ctx context.Context, request *registration.ListAllEntriesRequest) (_ *registration.ListAllEntriesResponse, err error) {
+	counter := telemetry_registrationapi.StartListEntriesCall(h.Metrics)
+	telemetry_common.AddCallerID(counter, getCallerID(ctx))
+	defer counter.Done(&err)
+	log := h.Log.WithField(telemetry.Method, telemetry.ListAllEntriesWithPages)
+
+	ds := h.getDataStore()
+	if request.PageSize == 0 {
+		request.PageSize = _defaultListEntriesPageSize
+	}
+	fetchResponse, err := ds.ListRegistrationEntries(ctx, &datastore.ListRegistrationEntriesRequest{
+		Pagination: &datastore.Pagination{
+			Token:    request.Token,
+			PageSize: request.PageSize,
+		},
+	})
+	if err != nil {
+		log.WithError(err).Error("Error trying to fetch entries")
+		return nil, status.Errorf(codes.Internal, "error trying to fetch entries: %v", err)
+	}
+	return &registration.ListAllEntriesResponse{
+		Entries: fetchResponse.Entries,
+		Pagination: &registration.Pagination{
+			Token:    fetchResponse.Pagination.Token,
```

Comment:
I'm a little worried about exporting the paging token directly from the datastore, not because it is private per-se, but because it doesn't allow for us to provide additional context in a backwards compatible way. One thing that that others do (i.e. Google APIs) is return a base64 encoded protobuf as their paging token. That allows some flexibility when introducing future changes.

Curious to hear @evan2645's thoughts on this.

Reply:
I can't help but wonder if the added complexity of an encoded protobuf is worth the effort. We can always do something fancier with v2 if we need to - that's probably a separate conversation though. 🤷‍♀